xilem_web/src/pod.rs

```diff
@@ -0,0 +1,219 @@
+            parent.into(),
+            was_removed,
+        )
+    }
+}
+
+impl<N: DomNode> ViewElement for Pod<N> {
+    type Mut<'a> = PodMut<'a, N>;
+}
+
+impl<N: DomNode> SuperElement<Pod<N>, ViewCtx> for AnyPod {
+    fn upcast(_ctx: &mut ViewCtx, child: Pod<N>) -> Self {
+        Pod::into_any_pod(child)
+    }
+
+    fn with_downcast_val<R>(
+        mut this: Self::Mut<'_>,
+        f: impl FnOnce(PodMut<N>) -> R,
+    ) -> (Self::Mut<'_>, R) {
+        let downcast = this.downcast();
+        let ret = f(downcast);
+        (this, ret)
+    }
+}
+
+impl<N: DomNode> AnyElement<Pod<N>, ViewCtx> for AnyPod {
+    fn replace_inner(this: Self::Mut<'_>, mut child: Pod<N>) -> Self::Mut<'_> {
+        child.apply_changes();
+        if let Some(parent) = this.parent {
+            parent
+                .replace_child(child.node.as_ref(), this.node.as_ref())
+                .unwrap_throw();
+        }
+        *this.node = Box::new(child.node);
+        *this.props = Box::new(child.props);
+        *this.flags = child.flags;
+        this
+    }
+}
+
+/// The mutable representation of [`Pod`].
+///
+/// This is a container which contains info of what has changed and provides mutable access to the underlying element and its props
+/// When it's dropped all changes are applied to the underlying DOM node
+pub struct PodMut<'a, N: DomNode> {
+    pub node: &'a mut N,
+    pub props: &'a mut N::Props,
+    pub flags: &'a mut PodFlags,
+    pub parent: Option<&'a web_sys::Node>,
+    pub was_removed: bool,
+    pub is_reborrow: bool,
+}
+
+impl<'a, N: DomNode> PodMut<'a, N> {
+    pub fn new(
+        node: &'a mut N,
+        props: &'a mut N::Props,
+        flags: &'a mut PodFlags,
+        parent: Option<&'a web_sys::Node>,
+        was_removed: bool,
+    ) -> PodMut<'a, N> {
+        PodMut {
+            node,
+            props,
+            flags,
+            parent,
+            was_removed,
+            is_reborrow: false,
+        }
+    }
+
+    pub fn reborrow_mut(&mut self) -> PodMut<N> {
+        PodMut {
+            node: self.node,
+            props: self.props,
+            flags: self.flags,
+            parent: self.parent,
+            was_removed: self.was_removed,
+            is_reborrow: true,
+        }
+    }
+
+    pub(crate) fn apply_changes(&mut self) {
+        if self.flags.needs_update() {
+            self.node.apply_props(self.props, self.flags);
+        }
+        self.flags.clear();
+    }
+}
+
+impl PodMut<'_, Box<dyn AnyNode>> {
+    fn downcast<N: DomNode>(&mut self) -> PodMut<N> {
+        PodMut::new(
+            self.node.deref_mut().as_any_mut().downcast_mut().unwrap(),
+            self.props.downcast_mut().unwrap(),
+            self.flags,
+            self.parent,
+            false,
+        )
+    }
+}
+
+impl<N: DomNode> Drop for PodMut<'_, N> {
+    fn drop(&mut self) {
+        if !self.is_reborrow && !self.was_removed {
+            self.apply_changes();
+        }
```

Comment:
maybe avoid `!` for better readability?
```suggestion
        if self.is_reborrow || self.was_removed {
            return;
        }
        self.apply_changes();
```